lib/vsc/filesystem/quota/tools.py

```diff
@@ -199,15 +199,23 @@ def process_user_quota(storage, gpfs, storage_name, filesystem, quota_map, user_
             if user_institute != institute:
                 continue
 
-            user_name = user_map.get(int(user_id), None)
-            if not user_name:
+            try:
+                user_name = user_map.get(user_id, pwd.getpwuid(int(user_id)).pw_name)
+            except KeyError:
+                logging.debug("Cannot find user ID %s in the user mapping", user_id)
+                continue
+            if not user_name.startswith('vsc'):
                 continue
 
+            if user_id not in user_map:
```

Comment:
move on try block, is also usefull for non-vsc lookups i guess

Reply:
We already continued and skipped the rest of the iteration if the user name doe not start with vsc.

Reply:
ping @stdweird I do not grok what you meant ^

Reply:
adding the resolved userid to the map/cache is also relevant for non-vsc users. so you can this earlier. perhaps not in the try block, but before previous `if` statement

Reply:
I fail to see why. We only store quota for VSC users, right?

Reply:
the cache is only to speed up the lookups, no need to have "slower" lookups for users even if we don't care about the data